v-next/hardhat/src/internal/builtin-plugins/solidity-test/helpers.ts

```diff
@@ -1,66 +1,119 @@
-import type { ArtifactsManager } from "../../../types/artifacts.js";
-import type { Artifact } from "@ignored/edr";
+import type {
+  Artifact as HardhatArtifact,
+  BuildInfo,
+} from "../../../types/artifacts.js";
+import type {
+  CompilationJobCreationError,
+  FailedFileBuildResult,
+  FileBuildResult,
+} from "../../../types/solidity/build-system.js";
+import type {
+  ArtifactId as EdrArtifactId,
+  Artifact as EdrArtifact,
+} from "@ignored/edr";
 
-import { HardhatError } from "@ignored/hardhat-vnext-errors";
-import { exists } from "@ignored/hardhat-vnext-utils/fs";
-import { resolveFromRoot } from "@ignored/hardhat-vnext-utils/path";
+import path from "node:path";
 
-export async function getArtifacts(
-  hardhatArtifacts: ArtifactsManager,
-): Promise<Artifact[]> {
-  const fqns = await hardhatArtifacts.getAllFullyQualifiedNames();
-  const artifacts: Artifact[] = [];
+import { HardhatError } from "@ignored/hardhat-vnext-errors";
+import { readJsonFile } from "@ignored/hardhat-vnext-utils/fs";
 
-  for (const fqn of fqns) {
-    const hardhatArtifact = await hardhatArtifacts.readArtifact(fqn);
-    const buildInfo = await hardhatArtifacts.getBuildInfo(fqn);
+import { FileBuildResultType } from "../../../types/solidity/build-system.js";
 
-    if (buildInfo === undefined) {
-      throw new HardhatError(
-        HardhatError.ERRORS.SOLIDITY_TESTS.BUILD_INFO_NOT_FOUND_FOR_CONTRACT,
-        {
-          fqn,
-        },
-      );
-    }
+type SolidityBuildResults =
+  | Map<string, FileBuildResult>
+  | CompilationJobCreationError;
+type SuccessfulSolidityBuildResults = Map<
+  string,
+  Exclude<FileBuildResult, FailedFileBuildResult>
+>;
 
-    const id = {
-      name: hardhatArtifact.contractName,
-      solcVersion: buildInfo.solcVersion,
-      source: hardhatArtifact.sourceName,
-    };
+/**
+ * This function asserts that the given Solidity build results are successful.
+ * It throws a HardhatError if the build results indicate that the compilation
+ * job failed.
+ */
+export function throwIfSolidityBuildFailed(
```

Comment:
For a future PR: we should use the same implementation here and during `compile`.

Reply:
mostly to have a consistent ui